lib/Backend/GlobOpt.cpp

```diff
@@ -8909,21 +8910,30 @@ GlobOpt::TypeSpecialization(
             {
                 return instr;
             }
+
+            // need to run typespec if cannot const fold InlineMathAbs, or dst will be dead
+            if (instr->m_opcode != Js::OpCode::InlineMathAbs)
+            {
+                runTypeSpec = false;
```

Comment:
This is still necessary because user can specify `-off:constfold` to ignore constant folding so no JIT exception in that way. Still need to handle that.